### PR TITLE
[R-package] fixed evaluation on valids in lightgbm() (fixes #2915)

### LINKED ISSUE #2915: [R-package] lightgbm() cannot evaluate metrics on validation sets
Evaluation on validation sets using the function `lightgbm()` is currently broken. [The documentation](https://github.com/microsoft/LightGBM/blob/master/R-package/R/lightgbm.R#L29) suggests that you should be able to pass `valids` to this function similar to the way it is passed to `lgb.train()`.

I discovered this working on #2912. As long as this bug exists, early stopping with the `lightgbm()` function is also broken.

## Environment info

Operating System: macOS Mojave 10.14.6

CPU/GPU model: 2.3 GHz Intel Core i5

C++/Python/R version: R 3.6.1
 
LightGBM version or commit hash: https://github.com/microsoft/LightGBM/commit/4aa888c8154301ebb6eb8730ec2dc1b58396497b

## Error message

Running the code in "Reproducible examples" yields the following error:

> Error in lgb.train(params = params, data = dtrain, nrounds = nrounds,  : 
  formal argument "valids" matched by multiple actual arguments

## Reproducible examples

```r
data(agaricus.train, package = "lightgbm")
data(agaricus.test, package = "lightgbm")
train <- agaricus.train
test <- agaricus.test
dvalid1 <- lgb.Dataset(
    data = train$data
    , labels = train$label
  )
  dvalid2 <- lgb.Dataset(
    data = train$data
    , labels = train$label
  )
  nrounds <- 10L
  bst <- lightgbm(
    data = train$data
    , label = train$label
    , num_leaves = 5L
    , nrounds = nrounds
    , objective = "binary"
    , metric = "binary_error"
    , valids = list(
      "valid1" = dvalid1
      , "valid2" = dvalid2
    )
  )
```

### PR DESCRIPTION
**Pull request by jameslamb**
See #2915 for details. Getting evaluation on validation sets with the `lightgbm()` function is currently broken. In this PR, I propose a fix and a test to prevent us from re-introducing the bug.

### CODE REVIEW

File: R-package/R/lightgbm.R [hunk 1]
```diff
@@ -73,24 +73,31 @@ lightgbm <- function(data,
...
 
+  train_args <- list(
+    "params" = params
+    , "data" = dtrain
+    , "nrounds" = nrounds
+    , "verbose" = verbose
+    , "eval_freq" = eval_freq
+    , "early_stopping_rounds" = early_stopping_rounds
+    , "init_model" = init_model
+    , "callbacks" = callbacks
```
Comment (guolinke) [inline R-package/R/lightgbm.R]: is `...` missed?

File: R-package/R/lightgbm.R [hunk 2]
```diff
@@ -73,24 +73,31 @@ lightgbm <- function(data,
...
-    params = params
-    , data = dtrain
-    , nrounds = nrounds
-    , valids = valids
-    , verbose = verbose
-    , eval_freq = eval_freq
-    , early_stopping_rounds = early_stopping_rounds
-    , init_model = init_model
-    , callbacks = callbacks
-    , ...
```
Comment (guolinke) [inline R-package/R/lightgbm.R]: is ... missed?

File: R-package/R/lightgbm.R (hunk 1 above)
Comment (jameslamb) [reply R-package/R/lightgbm.R]: ah, yes! Good catch. I'll add that and a test on it

Comment (jameslamb) [reply R-package/R/lightgbm.R]: oh wait @guolinke no I have it!

```
train_args <- append(train_args, list(...))
```

Comment (jameslamb) [reply R-package/R/lightgbm.R]: and it's being tested already because in my tests I have `metric = "binary_error"` (and `metric`  is not a keyword argument of the `lightgbm()` function)
